src/main/java/com/github/checkstyle/regression/Main.java

```diff
@@ -34,6 +74,169 @@ private Main() {
+        final Option branch = Option.builder("p")
+                .longOpt(OPT_PATCH_BRANCH)
+                .required()
+                .hasArg()
+                .desc("the name of the PR branch")
+                .build();
+        options.addOption(branch);
+
+        final Option tester = Option.builder("t")
+                .longOpt(OPT_CHECKSTYLE_TESTER_PATH)
+                .required(false)
+                .hasArg()
+                .desc("the path of the checkstyle-tester directory")
+                .build();
+        options.addOption(tester);
+
+        final Option stopAfterGenerateConfig = Option.builder()
+                .longOpt(OPT_STOP_AFTER_GENERATE_CONFIG)
+                .required(false)
+                .hasArg(false)
+                .desc("indicates that regression tool would stop after generating config")
+                .build();
+        options.addOption(stopAfterGenerateConfig);
+
+        return options;
+    }
+
+    /**
+     * Validates the parsed CLI arguments.
+     * @param args the parsed CLI arguments.
+     * @throws IllegalArgumentException the arguments are invalid
+     */
+    private static void validateArguments(Arguments args) {
+        if (!existAndIsDirectory(args.repoPath())) {
+            throw new IllegalArgumentException(
+                    "path of local git repo must exist and be a directory");
+        }
+        if (!args.stopAfterGenerateConfig()) {
+            if (args.checkstyleTesterPath().isPresent()) {
+                if (!existAndIsDirectory(args.checkstyleTesterPath().get())) {
+                    throw new IllegalArgumentException(
+                            "path of checkstyle tester must exist and be a directory");
+                }
+            }
+            else {
+                throw new IllegalArgumentException("missing checkstyleTesterPath, "
+                        + "which is required if you are using --stopAfterGenerateConfig mode");
+            }
+        }
+    }
+
+    /**
+     * Runs the regression tool.
+     * @param args the parsed CLI arguments.
+     * @throws Exception excute failure
+     */
+    private static void runRegression(Arguments args) throws Exception {
+        final File config = generateConfig(args.repoPath(), args.branch());
+        System.out.println("config generated at " + config.getAbsolutePath());
+        if (!args.stopAfterGenerateConfig()) {
+            final File report = ReportGenerator.generate(
+                    args.checkstyleTesterPath().get(), args.repoPath(), args.branch(), config);
+            System.out.println("report generated at " + report.getAbsolutePath());
+        }
+    }
+
+    /**
+     * Generates the config file.
+     * @param repoPath the path of the checkstyle repository
+     * @param branch   the name of the PR branch
+     * @return the generated config file
+     * @throws Exception generation failure
+     */
+    private static File generateConfig(String repoPath, String branch)
```

Comment:
Even though we don't need the entire arguments class, I sorta like sending it better than its pieces. Less documentation and easier to grab new arguments that we add.
What do you think about this?